File: solutions/javascript/MergekSortedLists/MergekSortedLists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode[]} lists
 * @return {ListNode}
 */
var mergeKLists = function(lists) {
    var a = [];
    for (var i = 0; i < lists.length; i++) {
        var l = lists[i];
        while (true) {
            if (!l) {
                break;
            }
            a.push(l.val);
            l = l.next;
        }
    }
    a = a.sort((x,y)=>x-y);
    if (a[0] == undefined) {
        return null;
    }
    var res = new ListNode(a[0]);
    var p = res;
    for (var j = 1; j < a.length; j++) {
        p.next = new ListNode(a[j]);
        p = p.next;
    }
    return res;
};

File: solutions/java/MergekSortedLists/MergekSortedLists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        List<Integer> a = new ArrayList<>();
        for (int i = 0; i < lists.length; i++) {
            ListNode l = lists[i];
            while (null != l) {
                a.add(l.val);
                l = l.next;
            }
        }
        Collections.sort(a);
        if (a.isEmpty()) {
            return null;
        }
        ListNode res = new ListNode(a.get(0));
        ListNode p = res;
        for (int j = 1; j < a.size(); j++) {
            p.next = new ListNode(a.get(j));
            p = p.next;
        }
        return res;
    }
}